### DISCUSSION TITLE
Intermittent flashing issue

### DISCUSSION THREAD

**Discussion by eob2000** [earlephilhower/arduino-pico#1346] [General]
Hi all,

Currently porting over some code from an AVR nano to a Pico. Porting went smoothly and I was able to get the code compiling and flashing pretty easily.

I have begun to hit a random flash issue. Most of the time the flash goes normal. I can choose the device COM port, hit flash in the IDE and away we go. Sketch runs perfectly and behaves as expected. However some of the time, the flash will not complete right, the IDE never completes flashing (The 'Uploading...' box never dismisses) and then when the Pico reboots windows throws up an unrecognised USB error (and the sketch is not running, I have a constant serial output on Serial1 so I can check if the sketch is doing anything). I can then boot the pico into BOOTSEL and flash it that way, which will then sometimes work, but other times the issue persists, eventually it will just begin working again. Sometimes when one of my Picos fails to flash my other Pico will flash fine, reverting to a known working version of the sketch seems to make no difference. It will eventually just randomly fix itself and start working with no clear reason why (No changes to the sketch etc)

Flashing a simple blank or hello world sketch usually works fine. flashing other U2F files (My go-to test is circuitpython) works perfectly and the pico behaves as expected, it is just when trying to do stuff related to this particular sketch that I encounter issues. However the fact that it is random and sometimes works perfectly has me confused. I also cannot see any clear way to troubleshoot this. Note I encounter the same issue on multiple windows machines, I have yet to try other operating systems.

Does anyone have any ideas?

arduino-pico version: 3.1.0
Arduino IDE version: 2.0.4
OS Version: Windows 10 21H2
RP2040 HW: Pico and Pico W

**Comment by earlephilhower** [comment]
On the Pico, the USB is run in the core itself and not another chip.  If your code is crashing, the USB port will die suddenly causing stuff like what you're seeing.

Alternatively, check the USB cables as those sometimes are less than stellar and can cause a voltage droop large enough to crash the chip or make even the ROM bootloader malfunction.

Finally, you can hook up a Picoprobe debugger and dump the state of the core to determine what exactly the chip is doing when it disappears.

**Reply by eob2000** [reply]
First of all, thanks for the reply and apologies for posting in the wrong place.

I did some more investigation and although I cannot say for certain since it was only ever intermittent, another USB cable seems to have done the trick. I had already tried 2 before! Looks like the 2040 is much more picky than other micros I've used before.

Thanks for the awesome work on this core!
